File: ITMO2022.Python/Practice02/Ex04/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Net.Mime.MediaTypeNames;

namespace DgemmCalc
{
    internal class Program
    {
        static void Main(string[] args)
        {
            DgemCalc calc = new DgemCalc();
            var matrix_a_double = calc.loadMatrix("matrix_a_float.csv");
            var matrix_b_double = calc.loadMatrix("matrix_b_float.csv");
            var matrix_c_double = calc.loadMatrix("matrix_c_float.csv");


            Console.WriteLine("Введите значение alpha");
            double alpha = Convert.ToDouble(Console.ReadLine());
            Console.WriteLine("Введите значение betta");
            double betta = Convert.ToDouble(Console.ReadLine());


            var watch = System.Diagnostics.Stopwatch.StartNew();
            var resultedMatrix = calc.dgemm_calcs(matrix_a_double, matrix_b_double, matrix_c_double, alpha, betta);
            watch.Stop();
            var elapsedMs = watch.ElapsedMilliseconds;
            Console.WriteLine(elapsedMs);

            var watch2 = System.Diagnostics.Stopwatch.StartNew();
            var resultedMatrix2 = calc.dgemm_calcs_multithreading(matrix_a_double, matrix_b_double, matrix_c_double, alpha, betta);
            watch.Stop();
            var elapsedMs2 = watch.ElapsedMilliseconds;
            Console.WriteLine(elapsedMs2);
        }
    }
}

File: ITMO2022.Python/Practice02/Ex04/Program.java
package DgemmCalc;

import java.util.Scanner;

public class Program {
    public static void main(String[] args) throws InterruptedException {
        DgemmCalc calc = new DgemmCalc();
        double[][] matrix_a_double = calc.loadMatrix("matrix_a_float.csv");
        double[][] matrix_b_double = calc.loadMatrix("matrix_b_float.csv");
        var matrix_c_double = calc.loadMatrix("matrix_c_float.csv");

        calc.matrices_multiply(matrix_a_double, matrix_b_double);

        Scanner sc= new Scanner(System.in);
        System.out.println("Введите значение alpha");
        double alpha = Double.parseDouble(sc.nextLine());
        System.out.println("Введите значение betta");
        double betta = Double.parseDouble(sc.nextLine());


        long startTime = System.currentTimeMillis();
        var resultedMatrix = calc.dgemm_calcs(matrix_a_double, matrix_b_double, matrix_c_double, alpha, betta);
        long endTime = System.currentTimeMillis();
        System.out.println(endTime - startTime);

        startTime = System.currentTimeMillis();
        var resultedMatrix2 = calc.dgemm_calcs_multithreading(matrix_a_double, matrix_b_double, matrix_c_double, alpha, betta);
        endTime = System.currentTimeMillis();
        System.out.println(endTime - startTime);

    }
}